### PR TITLE
Change current directory on environment reload request

### PR DESCRIPTION
**Pull request by AnatoliB**
Resolves issue #521

### CODE REVIEW

File: src/RequestProcessor.cs [hunk 1]
```diff
@@ -355,6 +356,11 @@ internal StreamingMessage ProcessFunctionEnvironmentReloadRequest(StreamingMessa
 
             var rpcLogger = new RpcLogger(_msgStream);
             rpcLogger.SetContext(request.RequestId, null);
+
+            var setCurrentDirMessage = string.Format(PowerShellWorkerStrings.SettingCurrentDirectory, environmentReloadRequest.FunctionAppDirectory);
+            rpcLogger.Log(isUserOnlyLog: false, LogLevel.Trace, setCurrentDirMessage);
+            Directory.SetCurrentDirectory(environmentReloadRequest.FunctionAppDirectory);
```
Comment (Francisco-Gamino) [inline src/RequestProcessor.cs]: Should we check that the directory exists before setting it? I remember that in the past for proxy call the working directory could be null.

**Comment by Francisco-Gamino** [comment]
One minor comment. 
Also, I don't remember if for updating protobuff files the we need a separate PR just for that, so please follow-up with @pragnagopa. Other than that, LGTM.

**Review by Francisco-Gamino — APPROVED** [review]
One minor comment.
Also, I don't remember if for updating protobuff files the we need a separate PR just for that, so please follow-up with @pragnagopa. Other than that, LGTM.

File: src/RequestProcessor.cs (hunk 1 above)
Comment (AnatoliB) [reply src/RequestProcessor.cs]: Checking for null: yes, thank you, done.
Checking that the directory exists: I don't know if we have a better way of handling the situation when it does not exist. Perhaps the best we can do is trust the host and fail if it does not exist - this is what NodeJs and Python workers also do.

File: src/RequestProcessor.cs [hunk 2]
```diff
@@ -355,6 +356,14 @@ internal StreamingMessage ProcessFunctionEnvironmentReloadRequest(StreamingMessa
 
             var rpcLogger = new RpcLogger(_msgStream);
             rpcLogger.SetContext(request.RequestId, null);
+
+            if (environmentReloadRequest.FunctionAppDirectory != null)
+            {
```
Comment (pragnagopa) [inline src/RequestProcessor.cs]: Please add unit test. Also, open issue here: https://github.com/Azure/azure-functions-integration-tests/issues/ - to track E2E test that verifies app directory after specialization.

**Review by pragnagopa — APPROVED** [review]
Thanks. LGTM. Added a minor comment on tests.

File: src/RequestProcessor.cs (hunk 1 above)
Comment (pragnagopa) [reply src/RequestProcessor.cs]: Yes. Trusting the host is OK. Any error handling should be done in  the host for this.

**Review by pragnagopa — APPROVED** [review]

File: src/RequestProcessor.cs (hunk 2 above)
Comment (AnatoliB) [reply src/RequestProcessor.cs]: Thank you. Added: https://github.com/Azure/azure-functions-integration-tests/issues/10

**Pull request merged**